templates.rst

```diff
@@ -0,0 +1,1061 @@
+should render with the subdirectory (e.g. ``/my_app/images/logo.png``). The
+``asset()`` function takes care of this by determining how your application is
+being used and generating the correct paths accordingly.
+
+.. tip::
+
+    The ``asset()`` function supports various cache busting techniques via the
+    :ref:`version <reference-framework-assets-version>`,
+    :ref:`version_format <reference-assets-version-format>`, and
+    :ref:`json_manifest_path <reference-assets-json-manifest-path>` configuration options.
+
+If you need absolute URLs for assets, use the ``absolute_url()`` Twig function
+as follows:
+
+.. code-block:: html+twig
+
+    <img src="{{ absolute_url(asset('images/logo.png')) }}" alt="Symfony!"/>
+
+.. tip::
+
+    Read the :doc:`Symfony's Webpack Encore documentation </frontend>` to learn
+    about integrating modern asset building features into Symfony applications.
+
+.. _twig-app-variable:
+
+The App Global Variable
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony creates for each request a context variable which is injected in every
+Twig template automatically as a variable called ``app``. It provides access to
+some application information:
+
+.. code-block:: html+twig
+
+    <p>Username: {{ app.user.username ?? 'Anonymous user' }}</p>
+    {% if app.debug %}
+        <p>Request method: {{ app.request.method }}</p>
+        <p>Application Environment: {{ app.environment }}</p>
+    {% endif %}
+
+The ``app`` variable (which is an instance of :class:`Symfony\\Bridge\\Twig\\AppVariable`)
+gives you access to these variables:
+
+``app.user``
+    The representation of the :ref:`current user <create-user-class>` or ``null``
+    if there is none. The value stored in this variable can be a
+    :class:`Symfony\\Component\\Security\\Core\\User\\UserInterface` object, any
+    other object which implements a ``__toString()`` method or even a regular string.
+``app.request``
+    The :class:`Symfony\\Component\\HttpFoundation\\Request` object that stores
+    the current :ref:`request data <accessing-request-data>` (depending on your
+    application, this can be a :ref:`sub-request <http-kernel-sub-requests>`
+    or a regular request).
+``app.session``
+    The :class:`Symfony\\Component\\HttpFoundation\\Session\\Session` object that
+    represents the current :doc:`user's session </sessions>` or ``null`` if there is none.
+``app.flashes``
+    An array of all the :ref:`flash messages <flash-messages>` stored in the session.
+    You can also get only the messages of some type (e.g. ``app.flashes('notice')``).
+``app.environment``
+    The name of the current :ref:`configuration environment <configuration-environments>`
+    (``dev``, ``prod``, etc).
+``app.debug``
+    True if in :ref:`debug mode <debug-mode>`. False otherwise.
+``app.token``
+    A :class:`Symfony\\Component\\Security\\Core\\Authentication\\Token\\TokenInterface`
+    object representing the security token.
```

Comment:
I wonder if we should just show an example of all of these in the above code block with a tiny Twig comment above each one - e.g.

```rst
{# app.request returns the Request object #}
Request method: {{ app.request.method }}
```

Reply:
I thought about that ... but `app.` gives you access to so many variables that the example could be complex (to show flash messages, security token, etc.)